### PR TITLE
Docs: Add docs for [plugin.plugin_id].tracing config key

### LINKED ISSUE #65629: Docs: Add docs for tracing
Add documentation on how to use tracing for backend plugins:

(Features added by)

https://github.com/grafana/grafana-plugin-sdk-go/pull/620

https://github.com/grafana/grafana/pull/63714

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Explains how to use `[plugin.plugin_id].tracing` config key for distributed tracing support in backend plugins.

**Why do we need this feature?**

To let grafana operators know how to enable tracing on a per-plugin basis.

**Who is this feature for?**

Grafana operators

**Which issue(s) does this PR fix?**:

Fixes #65629

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 1]
```diff
@@ -1965,6 +1965,18 @@ Enter a comma-separated list of plugin identifiers to hide in the plugin catalog
 
 <hr>
 
+## [plugin.plugin_id]
```
Comment (andresmgot) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1968]: maybe it's better to move this before `## [plugin.grafana-image-renderer]` (line 2023) to have all the [plugin.*] related sections close together

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 2]
```diff
@@ -1965,6 +1965,18 @@ Enter a comma-separated list of plugin identifiers to hide in the plugin catalog
...
+
+This section allows to configure plugin-specific settings. `plugin_id` must be replaced with the plugin id present in `plugin.json`.
```
Comment (andresmgot) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1970]: Maybe add something along the lines "properties described in this section are available for all plugins but it should be set independently"

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 3]
```diff
@@ -1965,6 +1965,18 @@ Enter a comma-separated list of plugin identifiers to hide in the plugin catalog
...
+
+### tracing
+
+> **Note**: Available in Grafana v9.5.0 or later, and [OpenTelemetry must be configured as well](#tracingopentelemetry).
+
+If `true`, propagate the tracing context to this backend plugin and enable tracing in the plugin (if it supports it).
```
Comment (andresmgot) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1976]:
```suggestion
If `true`, propagate the tracing context to the plugin backend and enable tracing (if it supports it).
```

**Review by andresmgot — COMMENTED** [review]
Just some minor suggestions. Letting @josmperez do the final round.

**Comment by xnyo** [comment]
Thanks for the review @andresmgot , I have addressed your suggestions :smile:

**Review by andresmgot — APPROVED** [review]
LGTM

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 4]
```diff
@@ -2020,6 +2020,20 @@ ha_engine_address = 127.0.0.1:6379
 
 <hr>
 
+## [plugin.plugin_id]
+
+This section allows to configure plugin-specific settings. `plugin_id` must be replaced with the plugin id present in `plugin.json`.
```
Comment (josmperez) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2025]:
```suggestion
This section explains how to configure plugin-specific settings. Replace thge `plugin_id` attribute with the plugin ID present in `plugin.json`.
```

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 5]
```diff
@@ -2020,6 +2020,20 @@ ha_engine_address = 127.0.0.1:6379
...
+
+Properties described in this section are available for all plugins, but must be set for each plugin.
```
Comment (josmperez) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2027]:
```suggestion
Properties described in this section are available for all plugins, but you must set them individually for each plugin.
```

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 6]
```diff
@@ -2020,6 +2020,20 @@ ha_engine_address = 127.0.0.1:6379
...
+
+### tracing
+
+> **Note**: Available in Grafana v9.5.0 or later, and [OpenTelemetry must be configured as well](#tracingopentelemetry).
+
+If `true`, propagate the tracing context to the plugin backend and enable tracing (if it supports it).
```
Comment (josmperez) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2033]:
```suggestion
If `true`, propagate the tracing context to the plugin backend and enable tracing (if the backend supports it).
```

**Review by josmperez — CHANGES_REQUESTED** [review]
Editorial suggestions.

File: docs/sources/setup-grafana/configure-grafana/_index.md (hunk 4 above)
Comment (xnyo) [reply docs/sources/setup-grafana/configure-grafana/_index.md]: I have committed the second part ("must be replaced" -> "replace the"), but I am not sure about the first one.

I was referring to the `[plugin.plugin_id]` ini file section, not the paragraph.

Do you think it's okay to leave it as "This section allows to configure plugin-specific settings" or maybe "This section can be used to configure plugin-specific settings" ?

Thanks 😄

Comment (josmperez) [reply docs/sources/setup-grafana/configure-grafana/_index.md]: Okay, I misunderstood the sentence. You might want to reword to "This section can be used..." or "This section of the of the ini file..."

**Review by josmperez — APPROVED** [review]

**Pull request merged**